packages/lib-sourcify/src/lib/solidityCompiler.ts

```diff
@@ -158,34 +186,25 @@ async function fetchSolcFromGitHub(
  *
  * @returns the requested solc instance
  */
-export function getSolcJs(version = 'latest'): Promise<any> {
+export async function getSolcJs(version = 'latest'): Promise<any> {
   // /^\d+\.\d+\.\d+\+commit\.[a-f0-9]{8}$/
   version = version.trim();
   if (version !== 'latest' && !version.startsWith('v')) {
     version = 'v' + version;
   }
 
-  const soljsonRepo = process.env.SOLJSON_REPO || 'soljson-repo';
-  const soljsonPath = path.resolve(soljsonRepo, `soljson-${version}.js`);
+  const soljsonRepo =
+    process.env.SOLJSON_REPO || path.join('/tmp', 'soljson-repo');
+  const fileName = `soljson-${version}.js`;
+  const soljsonPath = path.resolve(soljsonRepo, fileName);
 
-  if (fs.existsSync(soljsonPath)) {
+  if (!fs.existsSync(soljsonPath)) {
+    if (!(await fetchAndSaveSolc('bin', soljsonPath, version, fileName))) {
+      return false;
+    }
```

Comment:
<Repeating the comment>

If validation is useful in `solc-js` it would not serve the purpose here.

Think of the case: 
1. solcjs-v0.8.0 does not exist, so we fetch **and save** it
2. The validation in fetch function fails, so we return false
3. Another contract using v0.8.0 asks for the compiler
4. It exists on the disk so this time it won't return false